### PR TITLE
Run command for input

### PR DESCRIPTION
**Pull request by watson02**
# Integrating CLI

## Description

Integrating the functions from the Input and Output Teams into the CLI

## Type of Change

- [X] Feature
- [ ] Bug fix
- [ ] Documentation

## Contributors

- @watson02 

## Blocked by

- #65
- #68

## Reminder

 - All GitHub Actions should be in a passing state before any pull request is merged.
 - All PRs must be reviewed by at least one team member and one member of the Integration team!
 - Any issues this PR closes are tagged in the description!

### CODE REVIEW

File: gatorgrade/main.py
Comment (mariakimheinert) [inline gatorgrade/main.py:23]: In PR #68, it appears that the main function for the Output team is `run_and_display_command_checks`, whose parameter is not a file path. So, this function call should be updated to use the correct function and correct argument value.

File: gatorgrade/main.py
Comment (mariakimheinert) [inline gatorgrade/main.py:24]: The Generate team's main function should be called when the `gatorgrade generate` command is run. So, this function will need to be moved.

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
Hello @watson02! Thanks for writing this very important code! There are a few issues I've described below that must be addressed so that the code works correctly. Please let me know if you have any questions!

File: gatorgrade/main.py
Comment (mariakimheinert) [inline gatorgrade/main.py:23]: If you take a look at PR #68, you will see that the `run_and_display_command_checks` function does not take a file path as its argument; instead, it takes the list of checks that should be run and these checks are the output of the `parse_config` function. So, this code needs to be changed such that the argument passed to `run_and_display_command_checks` is the output of `parse_config`.

File: gatorgrade/main.py
Comment (mariakimheinert) [inline gatorgrade/main.py:39]: Unfortunately, `generate_config` has not been implemented yet. However, based on discussions I've had with the Generate team, the argument to the `generate_config` will likely not be the path to the `gatorgrade.yml` file. You should continue to communicate with the Generate team about the progress of the `generate_config` command!

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
Hello @watson02! This PR is definitely getting closer to being ready to merge! I have one change that you can make right now, and one that will need to be made once the Generate team implements their `generate_config` command. Please let me know if you have any questions!

File: gatorgrade/main.py
Comment (mariakimheinert) [inline gatorgrade/main.py:40]: The values `target_path_list` and `relative_run_path` are not defined here. `target_path_list` should be a list of strings captured through a command-line argument and `relative_run_path` does not need to be specified--we can just use its default value.
Additionally, the return value of `generate_config` probably does not need to be captured.

File: gatorgrade/main.py
Comment (mariakimheinert) [inline gatorgrade/main.py:23]: This is still not the correct argument for `run_and_display_command_checks`. The argument for `run_and_display_command_checks` should be the value returned by `parse_config`, which you store in the variable `checks`.

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
Hello @watson02, thanks for these revisions! This PR is really coming along, but there are still some problems that must be fixed, which I describe below. In addition to addressing the issues below, please also make sure to fix the failing GitHub Actions build. Let me know if you have any questions!

File: gatorgrade/main.py
Comment (Michionlion) [inline gatorgrade/main.py:23]: You should use the `filename` parameter here, instead of the hard-coded constant. Additionally, it looks like the default value for `filename` was reverted back to an older value -- it should be the value of `FILE`, not the string "FILE".

File: gatorgrade/main.py
Comment (Michionlion) [inline gatorgrade/main.py:39]: If `--force` isn't implemented, you should just remove the code related to it and instead directly call `generate_config` without any `if` statement.

**Review by Michionlion — CHANGES_REQUESTED** [review]
Hi @watson02! I've got two quick fixes for your code, after which I think things will be good to go. However, you'll still need to wait for #74 to be merged before this PR can be. Additionally, you'll want to manually run this (with something like `poetry run gatorgrader`) in both generate and run modes to check that your code works.

**Review by burgess01 — APPROVED** [review]

File: gatorgrade/main.py
Comment (Michionlion) [inline gatorgrade/main.py:5]: This is now `gatorgrade.input.parse_config`, I think.

**Review by Michionlion — CHANGES_REQUESTED** [review]
An import change that you'll find you need to make after merging in `main`. Please do that and then I think this will be good to merge!

**Review by Michionlion — DISMISSED** [review]
These changes look good! It looks like the GitHub Actions are failing because `poetry run task test` has some failing tests, which appear to be because the tests in `tests/test_main.py` are not correct tests for this new functionality.

**Review by laurennevill — DISMISSED** [review]
Good work!

**Review by mariakimheinert — COMMENTED** [review]
Thank you for implementing these changes, @watson02!

**Review by mariakimheinert — DISMISSED** [review]

**Review by burgess01 — APPROVED** [review]

**Review by mariakimheinert — APPROVED** [review]

**Review by laurennevill — APPROVED** [review]

**Review by Michionlion — APPROVED** [review]
This looks great @watson02! Thanks for fixing the tests, @burgess01!

**Pull request merged**
